java.hints/src/org/netbeans/modules/java/hints/suggestions/Lambda.java

```diff
@@ -238,6 +242,35 @@ public static ErrorDescription explicitParameterTypes(HintContext ctx) {
         return ErrorDescriptionFactory.forName(ctx, ctx.getPath(), Bundle.ERR_addExplicitLambdaParameters(), new AddExplicitLambdaParameterTypes(ctx.getInfo(), ctx.getPath()).toEditorFix());
     }
     
+    @Hint(displayName = "#DN_addVarLambdaParameters", description = "#DESC_addVarLambdaParameters", category = "suggestions", hintKind = Hint.Kind.ACTION, minSourceVersion = "11")
+    @Messages({
+        "DN_addVarLambdaParameters=Convert Lambda to Var Parameter Types",
+        "DESC_addVarLambdaParameters=Converts lambdas to var parameter types",
+        "ERR_addVarLambdaParameters=",
+        "FIX_addVarLambdaParameters=Use var parameter types"
+    })
+    @TriggerTreeKind(Kind.LAMBDA_EXPRESSION)
+    public static ErrorDescription implicitVarParameterTypes(HintContext ctx) {
+        // hint will be enable only for JDK-11 or above.
+        if (ctx.getInfo().getSourceVersion().compareTo(SourceVersion.RELEASE_9) < 2) {
+            return null;
+        }
+        // Check invalid lambda parameter declaration
+        if (ctx.getInfo().getTreeUtilities().isTreeHasError(ctx.getPath().getLeaf(), ERROR_CODES)) {
+            return null;
+        }
+        // Check var parameter types
+        LambdaExpressionTree let = (LambdaExpressionTree) ctx.getPath().getLeaf();
+        if (let.getParameters() != null && let.getParameters().size() > 0) {
+            VariableTree var = let.getParameters().get(0);
+            if (ctx.getInfo().getTreeUtilities().isVarType(TreePath.getPath(ctx.getPath(), var))) {
```

Comment:
Do not use `TreePath.getPath` to just append one item to the path (`ctx.getPath() + child`). `getPath` recursively searches through subtree. `new TreePath(parent, item)` is way more efficient.

Reply:
Use "new TreePath(parent, item)" method